app/models/track_sheet.rb

```diff
@@ -93,23 +93,19 @@ def generate_tracking_sheet(druid, pdf)
   # @param [Hash] doc Solr document or to_solr Hash
   # @return [Array<Array<String>>] Complex array suitable for pdf.table()
   def doc_to_table(solr_doc)
-    doc = solr_doc.with_indifferent_access # solr doc always has string keys, SolrDocument::FIELDS can be strings or symbols
+    solr_doc = SolrDocument.new(solr_doc) unless solr_doc.is_a?(SolrDocument)
```

Comment:
As far as I can tell, this method is only ever called here: https://github.com/sul-dlss/argo/blob/12018e207868fc6e8ef85f11c1d3f0aaca0899d9/app/models/track_sheet.rb#L63

Then `doc` is set to the return value of this method: https://github.com/sul-dlss/argo/blob/12018e207868fc6e8ef85f11c1d3f0aaca0899d9/app/models/track_sheet.rb#L123C11-L127

And that method only returns what comes back from `SearchService.query(%(id:"#{namespaced_druid}"), rows: 1)['response']['docs'].first` which as far as I can tell is only ever a hash. Can you confirm there's some context where a SolrDocument instance is passed to this method?

Reply:
Yeah, it was sort of a mix in the tests from what I could tell but a hash in the code. It's now getting a SolrDocument